jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -228,9 +228,56 @@ YUI.add('user-profile', function() {
     switchEnv: function(uuid, name) {
       var props = this.props;
       props.showConnectingMask();
+      this.close();
       props.switchModel(uuid, this.state.envList);
       props.dbEnvironmentSet('name', name);
+    },
+
+    /**
+      Set to a new, blank canvas with no model.
+
+      @method resetModel
+    */
+    resetModel: function() {
```

Comment:
So… having actually done the coding… I'm not sure it's really worth it. First off, this method really isn't reusable outside of `user-profile.js` because it's a wrapper around the more generic `utils.switchModel`. `utils.switchModel` is what other similar functionality should call. This function does a few other extra `UserProfile`-specific things like popping up the closing the profile.

The only function call that isn't `UserProfile`-specific is setting the model name in the DB. The problem there is that `this.db.evironment` is only in `app.js`, which the model name only goes down to the `switchModel` in `utils.js`. I think the right solution there is to pass `this.db.environment` into `utils.js`.

When all is said and done, there are two changes that seem worthwhile:

1. Combine `switchEnv` and `resetModel` into one function in `user-profile.js`.
2. Move the `dbEnvironmentSet` invocation into the utils' `switchModel`.